### PR TITLE
Add cBO (1D Problem)

### PR DESCRIPTION
**Pull request by arpanbiswas52**
Added notebook in example folder for 1D version. The 2D version is larger than 25 MB, so wont be able to upload. Please review. Thank you

### CODE REVIEW

**Review by ziatdinovmax — CHANGES_REQUESTED** [review]
A few comments:
- I noticed that you use ```train_x``` in standard BO and ```train_x_norm``` in cost BO. Is there a specific reason for that? It leads to different results at the first iteration for two algorithms. For proper comparison, one should probably use the same inputs.
- Multiple comments and docstrings refer to the multi-fidelity model, e.g.
```
def cost_acqfun(models, train_data, Data, cost_params, params, ieval, isnorm=False):
    """
    cost based acquisition function- EI based
    Args:
        models: Multi-fidelity and low-fidelity krigging model
```
 But I don't think we are using a multi-fidelity model here.

**Review by arpanbiswas52 — COMMENTED** [review]
Modified the analyses
Fixed the docstring

**Review by ziatdinovmax — CHANGES_REQUESTED** [review]
Thanks! Question about the structured BO part: you use a piecewise mean function with a "breaking point" t. However, the prior over t is a uniform distribution between 10 and 15, which is way outside of the train_x_norm range. Seems like this needs to be fixed. By the way, is the piecewise function really needed here?

**Comment by arpanbiswas52** [comment]
The piecewise function is just a demonstration to showcase how it will work with some incorrect information (which it worked still!). 
We can use different mean function as well.

On the prior distribution, the normal values are changed to real values and therefore the unif distribution is provided as part of the real parameter space (non-normalized one) [0, 15]. Please look below.

@jit
def mean_func(x: jnp.ndarray, params: Dict[str, float]) -> jnp.ndarray:

    x_data = x[:,0]
    #jax.debug.print("x_norm: {}", x_data)
    **lb = 0
    ub = 15
    x_data = x_data*(ub-lb) + lb**
    #jax.debug.print("x: {}", x_data)

    return jnp.piecewise(
            x_data, [x_data < params["t"], x_data >= params["t"]],
            [lambda x_data: (params["a2"]*jnp.sin(x_data*params["a1"])),
             lambda x_data: (params["b2"]*jnp.sin(x_data*params["b1"]))])

**Review by ziatdinovmax — APPROVED** [review]
Looks good. I will move it to examples/contrib, where I hope to gather examples of GPax applications for various "real-world" problems from different domains.

**Comment by arpanbiswas52** [comment]
Yes definitely, I am advising Shakti (Sergei’s postdoc) to apply cBO on nano indentation problem. We can use that notebook as real problem.
